sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezSpanProcessor.java

```diff
@@ -0,0 +1,208 @@
+   *
+   * @return a Collection of {@link io.opentelemetry.sdk.trace.ReadableSpan}.
+   */
+  public Collection<ReadableSpan> getCompletedSpans() {
+    Collection<ReadableSpan> completedSpans = new ArrayList<>();
+    synchronized (this) {
+      for (TracezSpanBuckets buckets : completedSpanCache.values()) {
+        completedSpans.addAll(buckets.getSpans());
+      }
+    }
+    return completedSpans;
+  }
+
+  /**
+   * Returns the completed span cache for {@link
+   * io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor}.
+   *
+   * @return a Map of String to {@link io.opentelemetry.sdk.extensions.zpages.TracezSpanBuckets}.
+   */
+  public Map<String, TracezSpanBuckets> getCompletedSpanCache() {
+    synchronized (this) {
+      return completedSpanCache;
+    }
+  }
+
+  /**
+   * Returns a new Builder for {@link io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor}.
+   *
+   * @return a new {@link io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor}.
+   */
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /** Builder class for {@link io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor}. */
+  public static final class Builder extends ConfigBuilder<Builder> {
+
+    private static final String KEY_SAMPLED = "otel.zpages.export.sampled";
+    private static final boolean DEFAULT_EXPORT_ONLY_SAMPLED = true;
+    private boolean sampled = DEFAULT_EXPORT_ONLY_SAMPLED;
+
+    private Builder() {}
+
+    /**
+     * Sets the configuration values from the given configuration map for only the available keys.
+     * This method looks for the following keys:
+     *
+     * <ul>
+     *   <li>{@code otel.zpages.export.sampled}: to set whether only sampled spans should be
+     *       exported.
+     * </ul>
+     *
+     * @param configMap {@link Map} holding the configuration values.
+     * @return this.
+     */
+    @VisibleForTesting
+    @Override
+    protected Builder fromConfigMap(
+        Map<String, String> configMap, NamingConvention namingConvention) {
+      configMap = namingConvention.normalize(configMap);
+      Boolean boolValue = getBooleanProperty(KEY_SAMPLED, configMap);
+      if (boolValue != null) {
+        return this.setExportOnlySampled(boolValue);
+      }
+      return this;
+    }
+
+    /**
+     * Sets whether only sampled spans should be exported.
+     *
+     * <p>Default value is {@code true}.
+     *
+     * @see Builder#DEFAULT_EXPORT_ONLY_SAMPLED
+     * @param sampled report only sampled spans.
+     * @return this.
+     */
+    public Builder setExportOnlySampled(boolean sampled) {
+      this.sampled = sampled;
+      return this;
+    }
+
+    /**
+     * Returns a new {@link io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor}.
+     *
+     * @return a new {@link io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor}.
+     */
+    public io.opentelemetry.sdk.extensions.zpages.TracezSpanProcessor build() {
```

Comment:
no need for a fully qualified name here. please import it.

Reply:
Done